commonv5/ts/prepare-task.ts

```diff
@@ -60,9 +65,34 @@ export default async function prepareTask(endpoint: Endpoint, rootPath: string)
 
   tl.setVariable(TaskVariables.SonarQubeScannerParams, jsonParams);
 
+  // Only do this if a version is explicitly set
+  if (cliVersion || msBuildVersion) {
+    const scannerPath = await downloadScanner(cliVersion, msBuildVersion);
+
+    tl.setVariable(TaskVariables.SonarQubeScannerLocation, scannerPath);
+  }
+  // we need to pass in the downloaded paths here
   await scanner.runPrepare();
 }
 
+async function downloadScanner(cliVersion: string, msBuildVersion: string) {
+  const cliVersionFileUrl = `https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${cliVersion}.zip`;
+  const msBuildFileUrl = `https://github.com/SonarSource/sonar-scanner-msbuild/releases/download/${msBuildVersion}/sonar-scanner-${msBuildVersion}-net.zip`;
+
+  const fileUrl = cliVersion ? cliVersionFileUrl : msBuildFileUrl;
+
+  tl.debug(`Downloading scanner from ${fileUrl}`);
+  const downloadPath = await toolLib.downloadTool(fileUrl);
+  tl.debug(`Downloaded: ${fileUrl} file to ${downloadPath}`);
+
+  tl.debug(`Extracting ${downloadPath}`);
+  const unzipPath = await toolLib.extractZip(downloadPath);
+  tl.debug(`Unzipped file to ${unzipPath}`);
+  // `downloadPath` now contains the path to the downloaded file
+
+  return unzipPath;
```

Comment:
Should we check whether a cached version of this Scanner CLI exists here, to avoid re-downloading on each pipeline run?